uniprotkb-rest/src/test/java/org/uniprot/api/uniprotkb/queue/UniProtKBAsyncDownloadIntegrationTest.java

```diff
@@ -77,14 +77,11 @@
 @TestInstance(TestInstance.Lifecycle.PER_CLASS)
 @WebMvcTest({UniProtKBDownloadController.class})
 @AutoConfigureWebClient
```

Comment:
move command code of and UniProtKBAsyncDownloadIntegrationTest and UnirefAsyncDownloadIntegration to a common class in common module. Use UniRef (R capital)

Reply:
**INITIAL STATE ON PR SUBMISSION**
UniProtKBAsyncDownloadIntegrationTest/UniRefAsyncDownloadIntegrationTest extends AbstractUniProtKBDownloadIT/AbstractUniRefDownloadIT
AbstractUniProtKBDownloadIT/AbstractUniRefDownloadIT extends AbstractStreamControllerIT
UniProtKBDownloadControllerIT/UniRefDownloadControllerIT extends AbstractDownloadControllerIT
AbstractDownloadControllerIT/AbstractDownloadControllerIT extends AbstractUniProtKBDownloadIT/AbstractUniRefDownloadIT
AbstractUniProtKBDownloadIT/AbstractUniRefDownloadIT extends AbstractStreamControllerIT
**AFTER 1ST ITERATION OF ABSTRACTION**
UniProtKBDownloadControllerIT/UniRefDownloadControllerIT extends AbstractDownloadControllerIT/AbstractDownloadControllerIT
AbstractDownloadControllerIT/AbstractDownloadControllerIT extends AbstractUniProtKBDownloadIT/AbstractUniRefDownloadIT
AbstractUniProtKBDownloadIT/AbstractUniRefDownloadIT extends AbstractDownloadIT
AbstractDownloadIT extends AbstractStreamControllerIT
UniProtKBAsyncDownloadIntegrationTest/UniRefAsyncDownloadIntegrationTest extends AbstractUniProtKBDownloadIT/AbstractUniRefDownloadIT
AbstractUniProtKBDownloadIT/AbstractUniRefDownloadIT extends AbstractDownloadIT
AbstractDownloadIT extends AbstractStreamControllerIT
**OBSERVATION**
I added one more layer of abstraction by introducing AbstractDownloadIT which abstracts helper functions and properties that are used by both uniRef & uniProtKB. There is a common dependency on AbstractUniProtKBDownloadIT/AbstractUniRefDownloadIT->AbstractDownloadIT by UniProtKBAsyncDownloadIntegrationTest/UniRefAsyncDownloadIntegrationTest & UniProtKBDownloadControllerIT/UniRefDownloadControllerIT which restricts us to further refactor tests in yet another or same abstract class due to project specific dependencies and tests replication in AsynchDownloadIntegrationTest & DownloadControllerIT